front/lib/api/files/upload.ts

```diff
@@ -502,12 +572,165 @@ export async function uploadToCloudStorage(
+      await file.markAsFailed();
+      return new Err({
+        name: "dust_error",
+        code: "internal_server_error",
+        message: `Failed to fetch conversation : ${r.error}`,
+      });
+    }
+
+    const conversation: ConversationType = r.value;
+
+    // Fetch the datasource linked to the conversation...
+    let dataSource = await DataSourceResource.fetchByConversationId(
+      auth,
+      conversation.id
+    );
+
+    if (!dataSource) {
+      // ...or create a new one.
+      const conversationsSpace =
+        await SpaceResource.fetchWorkspaceConversationsSpace(auth);
+
+      // IMPORTANT: never use the conversation sID in the name or description, as conversation sIDs are used as secrets to share the conversation within the workspace users.
+      const name = `Conversation ${uuidv4()}`;
+      const r = await createDataSourceWithoutProvider(auth, {
+        plan: auth.getNonNullablePlan(),
+        owner: auth.getNonNullableWorkspace(),
+        space: conversationsSpace,
+        name: name,
+        description: "Files uploaded to conversation",
+        conversation: conversation,
+      });
+
+      if (r.isErr()) {
+        await file.markAsFailed();
+        return new Err({
+          name: "dust_error",
+          code: "internal_server_error",
+          message: `Failed to create datasource : ${r.error}`,
+        });
+      }
+
+      dataSource = r.value.dataSource;
+    }
+
+    const documentId = file.sId; // Use the file id as the document id to make it easy to track the document back to the file.
+    const sourceUrl = file.getPublicUrl(auth);
+
+    // TODO(JIT) note, upsertDocument do not call runPostUpsertHooks (seems used for document tracker)
+    const upsertDocumentRes = await upsertDocument({
+      name: documentId,
+      source_url: sourceUrl,
+      text: content,
+      parents: [documentId],
+      tags: [`title:${file.fileName}`],
+      light_document_output: true,
+      dataSource,
+      auth,
+    });
+
+    if (upsertDocumentRes.isErr()) {
+      await file.markAsFailed();
+      return new Err({
+        name: "dust_error",
+        code: "internal_server_error",
+        message: "There was an error upserting the document.",
+        data_source_error: upsertDocumentRes.error,
+      });
+    }
+
+    const isTable = await isValidTableFile(file, content);
+
+    if (isTable) {
+      const tableId = file.sId; // Use the file sId as the table id to make it easy to track the table back to the file.
+      const upsertTableRes = await upsertTable({
+        tableId,
+        name: file.fileName,
+        description: "Table uploaded from file",
+        truncate: true,
+        csv: content,
+        tags: [`title:${file.fileName}`],
+        parents: [tableId],
+        async: false,
+        dataSource,
+        auth,
+        useAppForHeaderDetection: true,
+      });
+
+      if (upsertTableRes.isErr()) {
+        await file.markAsFailed();
+        return new Err({
```

Comment:
So we assume that a CSV file we fail to upsert as a table isn't valid? Seems bit harsh!